src/client.rs

```diff
@@ -515,7 +516,11 @@ pub(crate) unsafe extern "C" fn native_oauth_refresh_cb<C: ClientContext>(
 ) {
     // generate the token using generate_oauth_token
     let context = &mut *(opaque as *mut C);
-    let oauthbearer_config = CStr::from_ptr(oauthbearer_config).to_string_lossy();
+    let oauthbearer_config = match oauthbearer_config.is_null() {
```

Comment:
`oauthbearer_config` may be null if there is no configuration. This check protects against the resulting seg fault.